source/codegen/metadata_mutation.py

```diff
@@ -51,30 +51,32 @@ def get_size_param(param, parameters):
     named_params = { p['name'] : p for p in parameters }
     return named_params.get(param['size']['value'], None)
 
+def mark_mapped_enum_params(parameters, enums):
+    for param in (p for p in parameters if 'enum' in p):
+        enum_name = param['enum']
+        enum = enums[enum_name]
+        if 'generate-mappings' in enum:
+            del param['enum']
+            param['mapped-enum'] = enum_name
```

Comment:
Some of the downstream logic looks better to me if `mapped-enum` is a `bool` and `enum` is left alone. For example, there are a few blocks that do the same thing to `mapped-enum` or `enum` with a little extra for `mapped-enum`. 

Why is it better to swap the fields? Is this to support the case where both exist? Is that implemented with special metadata?

I think you cover some of this in the PR description. But this is very non-obvious. More of these conditionals/methods need comments on why they exist.